File: Exercise6/Position.java

class Position {

    private int x;
    private int y;

    public Position(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public String toString() {
        return ('A' + this.x - 1) + "" + this.y;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    @Override
    public boolean equals(Object other) {
        if (other == null) {
            return false;
        }
        if (!(other instanceof Position)) {
            return false;
        }
        Position op = (Position)other;
        return this.x == op.x && this.y == op.y;
    }
}

File: Exercise4/Position.cs
using System;

namespace Chess {

    [Serializable]
    internal class Position {

        int x;
        int y;
        string field;

        public string Field {

            get {
                return field;
            }

            set {
                field = value;
                x = value[0] - 'A' + 1;
                y = Int32.Parse(value.Substring(1));
            }

        }

        public int X {
            get {
                return x;
            }
        }

        public int Y {
            get {
                return y;
            }
        }

        public override string ToString() {
            return (char)(x - 1 + 'A') + "" + y;
        }

        public bool equals(Position other) {
            return x == other.x && y == other.y;
        }

        public Position(string field) {
            this.Field = field;
        }

    }
}
